File: python/136.只出现一次的数字.py
#
# @lc app=leetcode.cn id=136 lang=python3
#
# [136] 只出现一次的数字
#

# @lc code=start
class Solution:
    # 异或操作，一次遍历
    def singleNumber(self, nums: List[int]) -> int:
        ret = 0
        for _, num in enumerate(nums):
            ret ^= num
        return ret
# @lc code=end

File: cpp/136.只出现一次的数字.cpp
/*
 * @lc app=leetcode.cn id=136 lang=cpp
 *
 * [136] 只出现一次的数字
 */
#include <unordered_set>
#include <unordered_map>
#include <vector>
#include <algorithm>
using namespace std;
// @lc code=start
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        return method1(nums);
    }

    // 异或
    int method1(vector<int>& nums) {
        int ans = 0;
        for (int i = 0; i < nums.size(); i++) {
            ans ^= nums[i];
        }
        return ans;
    }
};
// @lc code=end

